composer.json

```diff
@@ -12,14 +12,20 @@
         }
     ],
     "require": {
-        "php": "^5.4|^7.0"
+        "php": "^7.0",
+        "psr/http-client": "^1.0",
+        "psr/http-factory": "^1.0"
     },
     "require-dev": {
         "phpunit/phpunit": "~4.0",
         "mockery/mockery": "~0.8",
-        "guzzlehttp/guzzle": "~5.0"
+        "guzzlehttp/guzzle": "~5.0",
+        "symfony/http-client": "^4.3",
+        "nyholm/psr7": "^1.1"
```

Comment:
since PSR18 doesn't deal with timeouts, let's remove the parts about PSR7/17/18 and focus on the symfony contracts